### PR TITLE
Renames for migrating to datastore v1beta3

### PR DESCRIPTION
**Pull request by ajkannan**
In addition to basic renames, there are a couple behavioral changes addressed in this PR:
- GqlQueryParameter no longer takes a name.
- The introduction of ArrayValue class, which is used in ListValue.java
- Removal of the "IsolationLevel" enum in TransactionOptions.java and TransactionImpl.java
- Removal of aggregation-related code in StructuredQuery.java

### CODE REVIEW

File: gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/BaseEntity.java [hunk 1]
```diff
@@ -90,10 +89,10 @@ private B self() {
     }
 
     @SuppressWarnings("unchecked")
-    protected B fill(DatastoreV1.Entity entityPb) {
+    protected B fill(com.google.datastore.v1beta3.Entity entityPb) {
       Map<String, Value<?>> copiedProperties = Maps.newHashMap();
-      for (DatastoreV1.Property property : entityPb.getPropertyList()) {
-        copiedProperties.put(property.getName(), Value.fromPb(property.getValue()));
+      for (Map.Entry<String, com.google.datastore.v1beta3.Value> property : entityPb.getProperties().entrySet()) {
```
Comment (aozarov) [inline gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/BaseEntity.java]: Fix indentation (here and bellow)

File: gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/BaseEntity.java [hunk 2]
```diff
@@ -377,21 +376,21 @@ public Blob getBlob(String name) {
...
   }
 
   protected abstract Builder<?, ?> emptyBuilder();
 
   @Override
-  protected final DatastoreV1.Entity toPb() {
-    DatastoreV1.Entity.Builder entityPb = DatastoreV1.Entity.newBuilder();
+  protected final com.google.datastore.v1beta3.Entity toPb() {
+    com.google.datastore.v1beta3.Entity.Builder entityPb = com.google.datastore.v1beta3.Entity.newBuilder();
+    Map<String, com.google.datastore.v1beta3.Value> propertiesPb = new HashMap<String, com.google.datastore.v1beta3.Value>();
```
Comment (aozarov) [inline gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/BaseEntity.java]: An alternative would be to call entityPb.getMutableProperties() and add directly.

File: gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/BaseKey.java [hunk 3]
```diff
@@ -172,20 +171,20 @@ public boolean equals(Object obj) {
   }
 
   @Override
-  protected DatastoreV1.Key toPb() {
-    DatastoreV1.Key.Builder keyPb = DatastoreV1.Key.newBuilder();
-    DatastoreV1.PartitionId.Builder partitionIdPb = DatastoreV1.PartitionId.newBuilder();
+  protected com.google.datastore.v1beta3.Key toPb() {
+    com.google.datastore.v1beta3.Key.Builder keyPb = com.google.datastore.v1beta3.Key.newBuilder();
+    com.google.datastore.v1beta3.PartitionId.Builder partitionIdPb = com.google.datastore.v1beta3.PartitionId.newBuilder();
```
Comment (aozarov) [inline gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/BaseKey.java]: Fix wrapping and indentation

File: gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/BatchImpl.java [hunk 4]
```diff
@@ -63,14 +62,14 @@
...
   public Batch.Response submit() {
     validateActive();
-    DatastoreV1.Mutation.Builder mutationPb = toMutationPb();
+    com.google.datastore.v1beta3.Mutation.Builder mutationPb = toMutationPb();
     if (force) {
       mutationPb.setForce(force);
     }
-    DatastoreV1.CommitRequest.Builder requestPb = DatastoreV1.CommitRequest.newBuilder();
-    requestPb.setMode(DatastoreV1.CommitRequest.Mode.NON_TRANSACTIONAL);
+    com.google.datastore.v1beta3.CommitRequest.Builder requestPb = com.google.datastore.v1beta3.CommitRequest.newBuilder();
```
Comment (aozarov) [inline gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/BatchImpl.java]: Fix wrapping

File: gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/Blob.java [hunk 5]
```diff
@@ -37,7 +36,7 @@
  *
  * @see <a href="https://cloud.google.com/datastore/docs/concepts/entities">Google Cloud Datastore Entities, Properties, and Keys</a>
  */
-public final class Blob extends Serializable<DatastoreV1.Value> {
+public final class Blob extends Serializable<Value> {
```
Comment (aozarov) [inline gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/Blob.java]: having a fully qualified Value would be better here.

File: gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/Cursor.java [hunk 6]
```diff
@@ -37,7 +36,7 @@
  * A Google Cloud Datastore cursor.
  * The cursor can be used to as a starting point or an ending point for a {@link Query}
  */
-public final class Cursor extends Serializable<DatastoreV1.Value> {
+public final class Cursor extends Serializable<Value> {
```
Comment (aozarov) [inline gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/Cursor.java]: keep Value fully qualified.

File: gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/DateTime.java [hunk 7]
```diff
@@ -34,7 +33,7 @@
  * @see <a href="https://cloud.google.com/datastore/docs/concepts/entities">Google Cloud Datastore
  *     Entities, Properties, and Keys</a>
  */
-public final class DateTime extends Serializable<DatastoreV1.Value>
+public final class DateTime extends Serializable<Value>
```
Comment (aozarov) [inline gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/DateTime.java]: here and in other places keep proto value fully qualified.

File: gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/GqlQuery.java [hunk 8]
```diff
@@ -142,15 +138,14 @@ public boolean equals(Object obj) {
...
-      return fromPb(DatastoreV1.GqlQueryArg.parseFrom(bytesPb));
+      return fromPb(com.google.datastore.v1beta3.GqlQueryParameter.parseFrom(bytesPb));
     }
 
-    static Binding fromPb(DatastoreV1.GqlQueryArg argPb) {
-      String name = argPb.hasName() ? argPb.getName() : null;
+    static Binding fromPb(com.google.datastore.v1beta3.GqlQueryParameter argPb) {
       if (argPb.hasCursor()) {
-        return new Binding(name, new Cursor(argPb.getCursor()));
+        return new Binding(null, new Cursor(argPb.getCursor()));
```
Comment (aozarov) [inline gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/GqlQuery.java]: this null does not look right to me.

File: gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/Query.java [hunk 9]
```diff
@@ -54,15 +53,15 @@
...
+        PB_TO_INSTANCE = Maps.newEnumMap(com.google.datastore.v1beta3.EntityResult.ResultType.class);
 
     static final ResultType<?> UNKNOWN = new ResultType<Object>(null, Object.class) {
 
       private static final long serialVersionUID = 1602329532153860907L;
 
-      @Override protected Object convert(DatastoreV1.Entity entityPb) {
-        if (entityPb.getPropertyCount() == 0) {
+      @Override protected Object convert(com.google.datastore.v1beta3.Entity entityPb) {
+        if (entityPb.getProperties().isEmpty() == 0) {
```
Comment (aozarov) [inline gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/Query.java]: isEmpty() returns boolean

File: gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/GqlQuery.java (hunk 8 above)
Comment (ajkannan) [inline gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/GqlQuery.java]: I meant to ask about this; a GqlQueryParameter no longer knows it's name, so I'm not quite sure how to populate "name".  The query now stores the name as the key in the map "namedBindings," but I don't see a way to get that information here.

File: gcloud-java-datastore/src/test/java/com/google/gcloud/datastore/LocalGcdHelper.java [hunk 10]
```diff
@@ -61,7 +61,7 @@
 
   public static final String DEFAULT_PROJECT_ID = "projectid1";
   public static final int PORT = 8080;
-  private static final String GCD = "gcd-v1beta2-rev1-2.1.2b";
+  private static final String GCD = "gcd-v1beta2-0.0.1";
```
Comment (ajkannan) [inline gcloud-java-datastore/src/test/java/com/google/gcloud/datastore/LocalGcdHelper.java]: Will change to gcd-v1beta3-0.0.1

File: gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/Cursor.java [hunk 11]
```diff
@@ -102,16 +100,16 @@ public static Cursor copyFrom(byte[] bytes) {
...
   @Override
   protected Object fromPb(byte[] bytesPb) throws InvalidProtocolBufferException {
-    return fromPb(DatastoreV1.Value.parseFrom(bytesPb));
+    return fromPb(com.google.datastore.v1beta3.Value.parseFrom(bytesPb));
   }
 
-  static Cursor fromPb(DatastoreV1.Value valuePb) {
-    return new Cursor(valuePb.getBlobValue());
+  static Cursor fromPb(com.google.datastore.v1beta3.Value valuePb) {
+    return new Cursor(com.google.datastore.v1beta3.valuePb.getBlobValue());
```
Comment (ajkannan) [inline gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/Cursor.java]: fixing to be `new Cursor(valuePb.getBlobValue())`

**Pull request closed without merging**